contracts/apps/AppProxyPinned.sol

```diff
@@ -21,7 +22,7 @@ contract AppProxyPinned is AppProxyBase {
         public // solium-disable-line visibility-first
     {
         setPinnedCode(getAppBase(_appId));
-        require(pinnedCode() != address(0));
+        require(isContract(pinnedCode()));
```

Comment:
Note that strictly, this isn't necessary as our `Kernel` doesn't let an app be set unless it's a contract (https://github.com/aragon/aragonOS/blob/dev/contracts/kernel/Kernel.sol#L177), so this is really more of an assertion (should we make it one?).

This also complicates the tests quite a bit, as you'll see in 961a7d7a81fcff0951225f95e72e8df697b5213d

Reply:
I don't think an assertion makes sense here, as a contract that follows the `IKernel` interface may not check whether a base is a contract. A require is fine IMO.